cmake/python_utils.cmake

```diff
@@ -0,0 +1,186 @@
+###
+# Finds the python binaries, libs, include, and site-packages paths
+#
+# The purpose of this file is to export variables that will be used in
+# kwiver/CMake/utils/kwiver-utils-python.cmake and
+# kwiver/sprokit/conf/sprokit-macro-python.cmake (the latter will eventually be
+# consolidated into the former)
+#
+# User options defined in this file:
+#
+#    MAJOR_PYTHON_VERSION
+#      The major python version to target (either 2 or 3)
+#
+#
+# Calls find_packages to on python interpreter/libraries which defines:
+#
+#    PYTHON_EXECUTABLE
+#    PYTHON_INCLUDE_DIR
+#    PYTHON_LIBRARY
+#    PYTHON_LIBRARY_DEBUG
+#
+# Exported variables used by python utility functions are:
+#
+#    PYTHON_VERSION
+#      the major/minor python version
+#
+#    PYTHON_ABI_FLAGS
+#      Python abstract binary interface flags (used internally for defining
+#      subsequent variables, but settable by the user as an advanced setting)
+#
+#    python_site_packages
+#      Location where python packages are installed relative to your python
+#      install directory. For example:
+#        Windows system install: Lib\site-packages
+#        Debian system install: lib/python2.7/dist-packages
+#        Debian virtualenv install: lib/python3.5/site-packages
+#
+#    python_sitename
+#      The basename of the python_site_packages directory. This is either
+#      site-packages (in most cases) or dist-packages (if your python was
+#      configured by a debian package manager). If you are using a python
+#      virtualenv (you should be) then this will be site-packages
+
+###
+# Private helper function to execute `python -c "<cmd>"`
+#
+# Runs a python command and populates an outvar with the result of stdout.
+# Be careful of indentation if `cmd` is multiline.
+#
+function(_pycmd outvar cmd)
```

Comment:
I don't believe this `python_utils` cmake module is actually used. We pull the _pycmd up into the main `CMakeLists.txt`, which would be the only piece from this file that we use.

Reply:
Yes, the utils are boilerplate that can be removed from pretty much all of these projects, I believe, because we don't check and branch on Python version during building.